File: lib_book/src/main/java/com/ebook/basebook/cache/ACache.java
package com.ebook.basebook.cache;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * 本地缓存   因本缓存只缓存书库主页 所以使用SP有条件可以替换成别的
 */
public class ACache {
    private final SharedPreferences preference;

    private ACache(Context ctx) {
        preference = ctx.getSharedPreferences("ACache", 0);
    }

    public static ACache get(Context ctx) {
        return new ACache(ctx);
    }

    public void put(String key, String value) {
        try {
            SharedPreferences.Editor editor = preference.edit();
            editor.putString(key, value);
            editor.apply();
        } catch (Exception ignored) {

        }
    }

    /**
     * 读取 String数据
     *
     * @return String 数据
     */
    public String getAsString(String key) {
        try {
            return preference.getString(key, null);
        } catch (Exception e) {
            return null;
        }
    }
}

File: lib_ebook_api/src/main/java/com/ebook/api/cache/ACache.kt
package com.ebook.api.cache

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

/**
 * 本地缓存   因本缓存只缓存书库主页 所以使用SP有条件可以替换成别的
 */
class ACache private constructor(context: Context) {
    private val preference: SharedPreferences = context.getSharedPreferences("ACache", 0)

    fun put(key: String, value: String) {
        try {
            preference.edit { putString(key, value) }
        } catch (_: Exception) {
        }
    }

    /**
     * 读取 String数据
     *
     * @return String 数据
     */
    fun getAsString(key: String): String? {
        return try {
            preference.getString(key, null)
        } catch (e: Exception) {
            null
        }
    }

    companion object {
        fun get(context: Context): ACache {
            return ACache(context)
        }
    }
}
